src/browser/modules/Stream/CypherFrame/PlanView.jsx

```diff
@@ -38,7 +38,14 @@ export class PlanView extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      extractedPlan: null
+      extractedPlan: null,
+      fullscreen: props.fullscreen
+    }
+  }
+
+  componentWillReceiveProps(nextProps, nextContext) {
+    if (nextProps.fullscreen !== this.props.fullscreen) {
+      this.setState({ fullscreen: nextProps.fullscreen })
```

Comment:
I don't see why fullscreen needs to be part of the state instead of just a prop?

Reply:
when click the fullscreen button props change but not update component but if it's be a state it will be update automatically.
![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/22843329/88896601-4e8fe280-d25f-11ea-9862-eb9d924f4200.gif)
as you can see in gif it's not updating when click fullscreen button but if be in fullscreen mode and click refresh button to recreate component it's work.
I'm not a react developer. maybe there is better way to fix this problem.

Reply:
Thanks for the video and your pr! I'll have a look at it! :)

Reply:
The problem is our `shouldComponentUpdate` method, it is missing a check `props.fullscreen !== this.props.fullscreen`. That's why the update doesn't go through! If you like you could rebase your branch on master and change the shouldComponentUpdate method and I'll merge your branch when the tests go through, or I can change it in one my branches 👍